src/main/java/wolox/training/models/User.java

```diff
@@ -59,23 +60,26 @@ public String getUsername() {
     }
 
     public void setUsername(String username) {
-        username = username;
+        Preconditions.checkNotNull(username);
+        this.username = username;
     }
 
     public String getName() {
         return name;
     }
 
     public void setName(String name) {
-        name = name;
+        Preconditions.checkNotNull(name);
```

Comment:
After checking if a field is null, check if the field is empty.

An error message can be added to each check :

```
Preconditions.checkNotNull(name, "The name field cannot be null.");
Preconditions.checkArgument(! name.isEmpty(), "The name field cannot be empty.");
this. name = name;
```

I recommend using a class to store error messages as constants.

Reply:
I put the messages and added it like constants.